frontend/CODE_GUIDELINES.md

```diff
@@ -0,0 +1,105 @@
+# Code Guidelines for React
+
+Hat tip to Clean Code by Robert C. Martin (Uncle Bob)
+
+## Goals
+
+- Consistency
+- Predictability
+- Maintainability
+- Readability
+
+## Style
+
+- Take time to craft great variable and function names. It's ok to be verbose. Avoid abbreviations.
+- Replace magic numbers with named constants.
+- Remove dead code aggressively. We can always recover it from version control.
+- Use vertical whitespace to aid readability.
+  - Related lines should be vertically close.
+  - A block of variable declarations should be followed by a blank line.
+  - Use a blank line above return statements.
+  - Don't stack code too tight vertically, especially `if` blocks.
+- Use Prettier/ESLint to avoid bike-shedding.
+- Prefer copy/paste to premature abstraction.
+  - Abstractions tend to reveal themselves quickly on the second or third copy/paste.
+  - It's easier to abstract code than to remove an abstraction.
+- Break switch statements into their own functions with returns rather than `break;`.
+
+## File structure
+
+- Higher-order functions go at the top of the file when possible.
+- Named exports are preferred to default exports. It's easier to control naming.
+- Inline exports are preferred to the define-then-export pattern.
+
+## Testing
+
+Flakey tests are the worst. The following patterns ensure that tests don't interact.
```

Comment:
Wonder if there is anything we should call out related to tests involving times and timezones. Run into quite a few flakey specs that start failing because a certain date was past or daylight savings occurred

Reply:
TIME ZONES!!!!!

I've had similar experiences.

This section of the doc was a reaction to some sketchy use of Jest. I don't use Jest as much anymore, but I know all the ways to NOT use it.